mobile/cmp/input/TextInput.scss

```diff
@@ -5,4 +5,15 @@
     height: 30px;
     padding: 0 var(--xh-pad-px);
   }
+
+  ons-input {
+    flex: 1;
+  }
```

Comment:
If we don't want to take this route of adding a rule to the ons-input element, we can set the default width to 100% when configuring the input in TextInput.js.

Reply:
will let tom comment on this

Reply:
I think using flex is the right approach - however, I think it would be better to use a BEM className here rather than relying on Onsen's component name.